src/apps/code-editor/src/app/components/Editor/components/MemoizedEditor/MemoizedEditor.js

```diff
@@ -29,6 +29,13 @@ export const MemoizedEditor = memo(
       ref.current.editor.setModel(model);
     }, [props.fileName]);
 
+    useEffect(() => {
+      if (!ref.current?.editor) return;
+      if (props.code !== ref.current.editor.getValue()) {
+        ref.current.editor.getModel().setValue(props.code);
+      }
+    }, [props.code]);
```

Comment:
As "code" changes are made in the Monaco editor those are broadcast to redux. Which would trigger a render as the value is changing. Eventually flowing to this point in the code. Would this not `setValue` on every key stroke? What performance implications does this have?

Reply:
@shrunyan Yea it would not set the Editor value on every keystroke. The check on line 34 ensures that the content of the editor only gets overwritten by new content in the redux store if and only if it is different to the current editor content. Therefore any other part of the app can modify the content value in the redux store and have it reflected onto the editor (without us making the Editor a fully controlled component)